tests/test_freshdesk_start_date.py

```diff
@@ -0,0 +1,166 @@
+
+        ##########################################################################
+        ### First Sync
+        ##########################################################################
+
+        # instantiate connection
+        conn_id_1 = connections.ensure_connection(self)
+
+        # run check mode
+        check_job_name_1 = self.run_and_verify_check_mode(conn_id_1)
+
+        # run initial sync
+        record_count_by_stream_1 = self.run_and_verify_sync(conn_id_1)
+        synced_records_1 = runner.get_records_from_target_output()
+
+        # Update based on sync data
+        first_sync_empty = self.test_streams - synced_records_1.keys()
+        if len(first_sync_empty) > 0:
+            print("Missing stream: {} in sync 1. Removing from test_streams. Add test data?".format(first_sync_empty))
+        first_sync_bonus = synced_records_1.keys() - self.test_streams
+        if len(first_sync_bonus) > 0:
+            print("Found stream: {} in first sync. Add to test_streams?".format(first_sync_bonus))
+        self.test_streams = self.test_streams - first_sync_empty
+
+        ##########################################################################
+        ### Update START DATE Between Syncs
+        ##########################################################################
+
+        print("REPLICATION START DATE CHANGE: {} ===>>> {} ".format(self.start_date, self.start_date_2))
+        self.start_date = self.start_date_2
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # create a new connection with the new start_date
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        check_job_name_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        # Update based on sync data
+        second_sync_empty = self.test_streams - synced_records_2.keys()
+        if len(second_sync_empty) > 0:
+            print("Missing stream(s): {} in sync 2. Updating expectations"\
+                  .format(second_sync_empty))
+            self.second_sync_empty = second_sync_empty
+        second_sync_bonus = synced_records_2.keys() - self.test_streams
+        if len(second_sync_bonus) > 0:
+            print("Found stream(s): {} in second sync. Add to test_streams?".format(second_sync_bonus))
+
+        for stream in test_streams:
+            with self.subTest(stream=stream):
+
+                if stream in self.second_sync_empty:
+                    print("No sync 2 data to compare for stream: {}, start_date obeyed".format(stream))
+
+                    continue
```

Comment:
This branch is fine, but we should have an assertion around the results.

Reply:
Updated.